app/src/main/java/ch/hikemate/app/utils/MapUtils.kt

```diff
@@ -379,15 +392,24 @@ object MapUtils {
    * @see [displayFacilities]
    */
   fun clearFacilities(mapView: MapView) {
-    mapView.overlays.removeAll { overlay ->
-      // The relatedObject was defined in displayFacilities for easier removal of the markers.
-      overlay is Marker && overlay.relatedObject == FACILITIES_RELATED_OBJECT_NAME
+    // Keep track of removed overlays to avoid concurrent modification
+    val overlaysToRemove =
+        mapView.overlays.filter { overlay ->
+          overlay is Marker && overlay.relatedObject == FACILITIES_RELATED_OBJECT_NAME
+        }
+
+    mapView.overlays.removeAll(overlaysToRemove)
+
+    // Force garbage collection of markers
+    overlaysToRemove.forEach { overlay ->
+      if (overlay is Marker) {
+        overlay.onDetach(mapView)
+      }
     }
 
-    // Trigger the map to be drawn again
+    // Clear the MapView's cache
     mapView.invalidate()
```

Comment:
Question: Same, why do we need this part?

Reply:
Same as above. It is to ensure that it is redrawn correctly.